server/src/main/java/org/elasticsearch/index/fielddata/fieldcomparator/LongValuesComparatorSource.java

```diff
@@ -128,4 +131,16 @@ protected long docValue() {
             }
         };
     }
+
+    @Override
+    public Object missingObject(Object missingValue, boolean reversed) {
+        if (targetNumericType == NumericType.DATE_NANOSECONDS) {
+            // special case to prevent negative values that would cause invalid nanosecond ranges
+            if (sortMissingFirst(missingValue) || sortMissingLast(missingValue)) {
+                final boolean min = sortMissingFirst(missingValue) ^ reversed;
+                return min ? 0L : Long.MAX_VALUE;
```

Comment:
We should use `DateUtils#MAX_NANOSECOND_INSTANT` as the max

Reply:
We need the converted nanoseconds value of that Instant though I think, which should be the same as Long.MAX_VALUE? I introduced another constant (https://github.com/elastic/elasticsearch/pull/74760/commits/8c6df4892fe8c7244f78289c796f06b7610888d0#diff-09ba5f62fc8de8923f01213874ef48657920b2cc1428ff4f27f5f06169ee8f1aR207) in my last commit to not have to do that coversion every time, just if you want to check.